test/support/supportedExtensions.ts

```diff
@@ -0,0 +1,32 @@
+/* tslint:disable max-line-length */
+import { languages } from '../../src/icon-manifest/languages';
+import {
+  IExtensionCollection,
+  IFileExtension,
+  FileFormat,
+  ISpecialFileCollection,
+  SpecialExtensionType,
+} from '../../src/models';
+
+export const extensions: (IExtensionCollection<IFileExtension> & ISpecialFileCollection) = {
+  special: {
```

Comment:
I didn't want to depend on the `icon` property in order to know what kind of default we're dealing with. Besides that, the `extension` property is completely irrelevant. We're dealing with a different type of model here and I wanted to make it evident. Not to just hack what we already have in order to make it work.

This structure simplifies the way we can apply merging for this kind of special files. If not, then we would have to use the current `extends` system and, as I said before, use the `icon` property value to determine what kind of `default icon` this is.

Anyway, this is a WIP merge, and I'd ask you to bear with me until is finished. I have to say that my first approach was to use the default API in order to make it work. But once I started to think about how I would have to alter the `iconGenerator` and the `manifestMerger`  I thought that making the API clearer would be a benefit in the long run.